lib/appsignal/config.ex

```diff
@@ -179,6 +181,9 @@ defmodule Appsignal.Config do
     System.put_env("APPSIGNAL_HOSTNAME", config[:hostname])
   end
 
+  defp app_name_to_string(name) when is_atom(name), do: Atom.to_string(name)
+  defp app_name_to_string(name), do: name
```

Comment:
`System.put_env/2` expects a string as the value, so passing everything through results in a crash that could be triggered earlier if we guard the second variant of this function with an `is_binary`, like this:

```elixir
defp app_name_to_string(name) when is_binary(name), do: name
```